code-challenges.js

```diff
@@ -13,21 +31,66 @@ var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+const sortedOdds = (array) => {
+  //use a filter to traverse through the array
+  // find the odd numbers first
+  // finally sort low to high
+  return array.filter(value => ((typeof value === 'number') && (value % 2 !== 0))).sort((a,b) => a-b)
+}
+
+console.log(sortedOdds(fullArr1))
+console.log(sortedOdds(fullArr2))
 
 
 
 // --------------------3) Create a function that takes in a string of a single word and returns the middle letter of the word. If the word is an even number of letters, return the two middle letters.
 
+const midLetter = (string) => {
+  // local array to hold temp values
+  let tempString = []
+  // split the word in the middle using .slice()
+  tempString = string.slice(Math.floor(string.length / 2) - 1)
+  // determine how many chars to return, 1 for odd, 2 for even
+  //console.log(tempString)
+  if (string.length % 2 === 0) {
+    // even
+    return tempString.slice(0,2)
+  } else {
+    // odd
+    return tempString.slice(1,2)
+  }
+}
+
 var middleLetters1 = "hello"
 // Expected output: “l”
 var middleLetters2 = "rhinoceros"
 // Expected output: “oc”
 
+console.log(midLetter(middleLetters1))
+console.log(midLetter(middleLetters2))
 
 
 
 // --------------------4) Create a CLASS to get the area of a sphere. Create three spheres with different radi as test cases. Area of a sphere =  4πr^2 (four pi r squared)
+class Sphere {
+  constructor() {
+    this.area = 0
+  }
 
+  findArea(radius) {
+    return this.area = 4 * 3.14 * radius**2
+  }
+}
+
+// create diff size spheres from the Sphere class
+var littleSphere = new Sphere()
+var mediumSphere = new Sphere()
+var largeSphere = new Sphere()
+
+// find the area of those spheres
+console.log("the area of the sphere is ", `${littleSphere.findArea(4).toFixed(2)}`, " Square units")
+console.log("the area of the sphere is ", `${mediumSphere.findArea(10).toFixed(2)}`, " Square units")
+console.log("the area of the sphere is ", `${largeSphere.findArea(23).toFixed(2)}`, " Square units")
```

Comment:
this solution does do the job! I think a small refactor of adding radius to the constructor would keep the user from having to pass an argument in the console.log, and instead could pass it when the object is created.